### PR TITLE
Add com.leinardi.gkraken

### PR DESCRIPTION
**Pull request by leinardi**
I'm moving this app from PyPI to Flathub. Info on the app here: https://gitlab.com/leinardi/gkraken

### CODE REVIEW

File: flatpak-pip-generator [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+__license__ = 'MIT'
+
```
Comment (bilelmoussaoui) [inline flatpak-pip-generator]: No need to ship this here.

File: refresh-pip-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+./flatpak-pip-generator --requirements-file=../requirements.txt --output pypi-dependencies
```
Comment (bilelmoussaoui) [inline refresh-pip-dependencies.sh]: Same as the other file.

**Comment by LeandroStanger** [comment]
bot, build com.leinardi.gkraken

File: com.leinardi.gkraken.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+  "app-id": "com.leinardi.gkraken",
+  "command": "gkraken",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.32",
```
Comment (bilelmoussaoui) [inline com.leinardi.gkraken.json]: Have you tried bumping this to 3.34?

Comment (LeandroStanger) [inline com.leinardi.gkraken.json]: Updat runtime to 3.34

Comment (leinardi) [reply com.leinardi.gkraken.json]: Not yet, I'll test it now and update if works fine.

Comment (leinardi) [reply com.leinardi.gkraken.json]: Build on 3.34 fails with this error:
```
Starting build of com.leinardi.gkraken
========================================================================
Building module libdbusmenu in /home/leinardi/Workspace/gitlab/gkraken/.flatpak-builder/build/libdbusmenu-1
========================================================================
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether NLS is requested... yes
./configure: line 3080: intltool-update: command not found
checking for intltool >= 0.35.0...  found
configure: error: Your intltool is too old.  You need intltool 0.35.0 or later.
Error: module libdbusmenu: Processo figlio uscito con codice 1
```

**Comment by leinardi** [comment]
I'm not sure why this is failing:

```
FB: Running: git config --bool core.bare false
FB: Running: git checkout 0.13.0
Note: checking out '0.13.0'.
You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.
If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:
  git checkout -b <new-branch-name>
HEAD is now at 6dbc4b1 Prepare for release 0.13.0
FB: Running: git rev-parse --verify --quiet 0.13.0:.gitmodules
FB: Running: git show 0.13.0:.gitmodules
FB: Running: git ls-tree 0.13.0 flatpak
FB: Running: git config submodule.flatpak.url file:///srv/buildbot/worker/build-i386-2/build/.flatpak-builder/git/https_github.com_leinardi_flathub.git
FB: Running: git submodule update --init flatpak
Cloning into '/srv/buildbot/worker/build-i386-2/build/.flatpak-builder/build/gkraken-1/flatpak'...
error: Server does not allow request for unadvertised object bb9385894b3ba670ed0a1ba62004d6b958e21213
Fetched in submodule path 'flatpak', but it did not contain bb9385894b3ba670ed0a1ba62004d6b958e21213. Direct fetching of that commit failed.
]2;flatpak-builder: Cleanup gkrakenError: module gkraken: Child process exited with code 1
```
It's because I have my fork as a submodule?

**Comment by leinardi** [comment]
bot, build com.leinardi.gkraken

File: com.leinardi.gkraken.json (hunk 3 above)
Comment (bilelmoussaoui) [reply com.leinardi.gkraken.json]: intltool was removed from Freedesktop 19.08. It's available from shared-modules. Just add that before libappindicator.

**Comment by leinardi** [comment]
bot, build com.leinardi.gkraken

File: com.leinardi.gkraken.json (hunk 3 above)
Comment (leinardi) [reply com.leinardi.gkraken.json]: Ok, I had to update the shared modules and add the `dbus-glib-0.110.json` before the `libappindicator-gtk3-introspection-12.10.json`.

File: refresh-pip-dependencies.sh (hunk 2 above)
Comment (leinardi) [reply refresh-pip-dependencies.sh]: removed

File: flatpak-pip-generator (hunk 1 above)
Comment (leinardi) [reply flatpak-pip-generator]: removed

**Comment by bilelmoussaoui** [comment]
I don't have any device to try it out but it seems to launch fine other than this
```
(gkraken:2): Gtk-WARNING **: 13:59:38.823: Could not load image 'gkraken.svg': The resource at “/com/leinardi/gkraken/ui/gkraken.svg” does not exist
ERROR: Err = Kraken USB interface error (check USB cable connection)
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/rx/core/observable/defer.py", line 29, in subscribe
    result = factory(scheduler)
  File "/app/lib/python3.7/site-packages/gkraken/interactor.py", line 52, in <lambda>
    return rx.defer(lambda _: rx.just(self._kraken_repository.get_status()))
  File "/app/lib/python3.7/site-packages/gkraken/util/concurrency.py", line 23, in synced_method
    return method(self, *args, **kws)
  File "/app/lib/python3.7/site-packages/gkraken/repository.py", line 48, in get_status
    self._load_driver()
  File "/app/lib/python3.7/site-packages/gkraken/repository.py", line 86, in _load_driver
    raise ValueError("Kraken USB interface error (check USB cable connection)")
ValueError: Kraken USB interface error (check USB cable connection)
ERROR: Refresh error: 
NoneType: None
ERROR: Check new version error: Expecting value: line 1 column 1 (char 0)
NoneType: None
```

**Comment by leinardi** [comment]
Just tried the bot test build and it seems to work fine.

**Comment by bilelmoussaoui** [comment]
Repository has been created: https://github.com/flathub/com.leinardi.gkraken


Thanks!

**Comment by bilelmoussaoui** [comment]
@leinardi Please confirm the write-access invitation. Could you track down the two errors I mentioned above? Thanks

**Comment by leinardi** [comment]
@bilelmoussaoui invitation accepted. One of the error is the version check that is looking for the latest release on https://flathub.org/apps/details/com.leinardi.gkraken but is not finding it (it wasn't published yet). I'll investigate the missing svg error.

**Comment by bilelmoussaoui** [comment]
It takes sometime for the app to appear on Flathub. You can see here https://flathub.org/builds/#/
That the app just got built https://flathub.org/builds/#/builders/32/builds/8913. The app is published after hours, you can either login with your github account and click on Publish or wait until it's published automatically. 
Once the application is published, it takes a few minutes for the website to refresh its cache and display it.

**Comment by leinardi** [comment]
I just started the manual publication, thanks!

**Pull request closed without merging**